CCS Cypress integration

cavokz opened the pull request:
## Summary

Add new Cypress test flow for CCS related Security Solution tests.

Meta https://github.com/elastic/security-team/issues/1301.
Supersedes https://github.com/elastic/kibana/pull/101773.

MadameSheema committed:
Add CCS Cypress test runner

cavokz committed:
Split flow for CCS Cypress tests

cavokz committed:
Make esArchiver load data onto the remote cluster

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/)
* Commit: 37b9069c871e6078f7fd19a84470a8e9821e965c
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/execution/node/428/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/","number":"135679","commit":"37b9069c871e6078f7fd19a84470a8e9821e965c"}]}
PIPELINE-->

cavokz committed:
Add CCS related documentation

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/)
* Commit: 75072ed6dd0be0c214e60da3fe350f69a15d1599
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/execution/node/428/log)

### History
* :broken_heart: [Build #135679](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/) failed 37b9069c871e6078f7fd19a84470a8e9821e965c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/","number":"135679","commit":"37b9069c871e6078f7fd19a84470a8e9821e965c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/","number":"135711","commit":"75072ed6dd0be0c214e60da3fe350f69a15d1599"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

cavokz committed:
Allow overriding @kbn/dev-utils's CA_CERT_PATH

integration-test/jenkins-test.sh uses its own CA, need to use that
otherwise esArchiver CLI would fail.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/)
* Commit: b142dc36ab404f0108c897bb874ce9b4b6f68343
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/execution/node/428/log)

### History
* :broken_heart: [Build #135711](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/) failed 75072ed6dd0be0c214e60da3fe350f69a15d1599
* :broken_heart: [Build #135679](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/) failed 37b9069c871e6078f7fd19a84470a8e9821e965c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/","number":"135679","commit":"37b9069c871e6078f7fd19a84470a8e9821e965c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/","number":"135711","commit":"75072ed6dd0be0c214e60da3fe350f69a15d1599"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/","number":"136332","commit":"b142dc36ab404f0108c897bb874ce9b4b6f68343"}]}
PIPELINE-->

MadameSheema commented on x-pack/plugins/security_solution/cypress/README.md:
```diff
@@ -115,8 +115,41 @@ cd x-pack/plugins/security_solution
+yarn kbn bootstrap
+
+# build the plugins/assets that cypress will execute against
+node scripts/build_kibana_platform_plugins
```
For a custom target, this step it is not needed :)

cavokz replied:
Not so sure, I ended up in such situation that I had to rebuild the plugins otherwise the test failed for the wrong reason.

cavokz replied:
Resolved

cavokz committed:
Add CCS specific rules with customizable remote name

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/)
* Commit: 627ea9b77540839edcc77159a160fa14d6526f25
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103941/627ea9b77540839edcc77159a160fa14d6526f25)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #136332](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/) failed b142dc36ab404f0108c897bb874ce9b4b6f68343
* :broken_heart: [Build #135711](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/) failed 75072ed6dd0be0c214e60da3fe350f69a15d1599
* :broken_heart: [Build #135679](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/) failed 37b9069c871e6078f7fd19a84470a8e9821e965c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/","number":"135679","commit":"37b9069c871e6078f7fd19a84470a8e9821e965c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/","number":"135711","commit":"75072ed6dd0be0c214e60da3fe350f69a15d1599"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/","number":"136332","commit":"b142dc36ab404f0108c897bb874ce9b4b6f68343"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/","number":"136382","commit":"627ea9b77540839edcc77159a160fa14d6526f25"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136391/)
* Commit: 8b021641bc1c35446181c1d9ebaf57e1160f407d
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103941/8b021641bc1c35446181c1d9ebaf57e1160f407d)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #136382](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/) succeeded 627ea9b77540839edcc77159a160fa14d6526f25
* :broken_heart: [Build #136332](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/) failed b142dc36ab404f0108c897bb874ce9b4b6f68343
* :broken_heart: [Build #135711](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/) failed 75072ed6dd0be0c214e60da3fe350f69a15d1599
* :broken_heart: [Build #135679](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/) failed 37b9069c871e6078f7fd19a84470a8e9821e965c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/","number":"135679","commit":"37b9069c871e6078f7fd19a84470a8e9821e965c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/","number":"135711","commit":"75072ed6dd0be0c214e60da3fe350f69a15d1599"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/","number":"136332","commit":"b142dc36ab404f0108c897bb874ce9b4b6f68343"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/","number":"136382","commit":"627ea9b77540839edcc77159a160fa14d6526f25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136391/","number":"136391","commit":"8b021641bc1c35446181c1d9ebaf57e1160f407d"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136438/)
* Commit: 10d4bdc71a93d03a2cb445afd6dc3407a022f453
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136438/execution/node/428/log)

### History
* :green_heart: [Build #136391](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136391/) succeeded 8b021641bc1c35446181c1d9ebaf57e1160f407d
* :green_heart: [Build #136382](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/) succeeded 627ea9b77540839edcc77159a160fa14d6526f25
* :broken_heart: [Build #136332](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/) failed b142dc36ab404f0108c897bb874ce9b4b6f68343
* :broken_heart: [Build #135711](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/) failed 75072ed6dd0be0c214e60da3fe350f69a15d1599
* :broken_heart: [Build #135679](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/) failed 37b9069c871e6078f7fd19a84470a8e9821e965c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135679/","number":"135679","commit":"37b9069c871e6078f7fd19a84470a8e9821e965c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/","number":"135711","commit":"75072ed6dd0be0c214e60da3fe350f69a15d1599"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/","number":"136332","commit":"b142dc36ab404f0108c897bb874ce9b4b6f68343"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/","number":"136382","commit":"627ea9b77540839edcc77159a160fa14d6526f25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136391/","number":"136391","commit":"8b021641bc1c35446181c1d9ebaf57e1160f407d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136438/","number":"136438","commit":"10d4bdc71a93d03a2cb445afd6dc3407a022f453"}]}
PIPELINE-->

MadameSheema reviewed (approved):
@cavokz this is more than awesome!! Lots of thanks for your hard work here :D 

The only little thing, before thinking about it, probably it is better to name the folder that contains  the tests as `ccs_integration` instead of `ccs`.

Again, lots of thanks for this!! :)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136441/)
* Commit: 62bab351f9a92a970d177f9d181a6423d47acb87
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103941/62bab351f9a92a970d177f9d181a6423d47acb87)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #136438](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136438/) failed 10d4bdc71a93d03a2cb445afd6dc3407a022f453
* :green_heart: [Build #136391](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136391/) succeeded 8b021641bc1c35446181c1d9ebaf57e1160f407d
* :green_heart: [Build #136382](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/) succeeded 627ea9b77540839edcc77159a160fa14d6526f25
* :broken_heart: [Build #136332](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/) failed b142dc36ab404f0108c897bb874ce9b4b6f68343
* :broken_heart: [Build #135711](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/) failed 75072ed6dd0be0c214e60da3fe350f69a15d1599

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135711/","number":"135711","commit":"75072ed6dd0be0c214e60da3fe350f69a15d1599"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/","number":"136332","commit":"b142dc36ab404f0108c897bb874ce9b4b6f68343"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/","number":"136382","commit":"627ea9b77540839edcc77159a160fa14d6526f25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136391/","number":"136391","commit":"8b021641bc1c35446181c1d9ebaf57e1160f407d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136438/","number":"136438","commit":"10d4bdc71a93d03a2cb445afd6dc3407a022f453"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136441/","number":"136441","commit":"62bab351f9a92a970d177f9d181a6423d47acb87"}]}
PIPELINE-->

cavokz commented:
> The only little thing, before thinking about it, probably it is better to name the folder that contains the tests as `ccs_integration` instead of `ccs`.

Done

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136490/)
* Commit: 985083c14302683b1571565558719e8b84c64e01
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103941/985083c14302683b1571565558719e8b84c64e01)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #136441](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136441/) succeeded 62bab351f9a92a970d177f9d181a6423d47acb87
* :broken_heart: [Build #136438](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136438/) failed 10d4bdc71a93d03a2cb445afd6dc3407a022f453
* :green_heart: [Build #136391](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136391/) succeeded 8b021641bc1c35446181c1d9ebaf57e1160f407d
* :green_heart: [Build #136382](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/) succeeded 627ea9b77540839edcc77159a160fa14d6526f25
* :broken_heart: [Build #136332](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/) failed b142dc36ab404f0108c897bb874ce9b4b6f68343

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136332/","number":"136332","commit":"b142dc36ab404f0108c897bb874ce9b4b6f68343"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/","number":"136382","commit":"627ea9b77540839edcc77159a160fa14d6526f25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136391/","number":"136391","commit":"8b021641bc1c35446181c1d9ebaf57e1160f407d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136438/","number":"136438","commit":"10d4bdc71a93d03a2cb445afd6dc3407a022f453"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136441/","number":"136441","commit":"62bab351f9a92a970d177f9d181a6423d47acb87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136490/","number":"136490","commit":"985083c14302683b1571565558719e8b84c64e01"}]}
PIPELINE-->

banderror commented on x-pack/plugins/security_solution/cypress/README.md:
```diff
@@ -208,6 +242,44 @@ Because of `cy.exec`, used to invoke `es_archiver`, it's necessary to override i
 
+### CCS
+
+Tests running in CCS configuration need to care about two aspects:
+
+1. data (eg. to trigger alerts) is generated/loaded on the remote cluster
+2. queries (eg. detection rules) refer to remote indices
```
By `data`, if we talk about detection engine, do you mean only source events here, or any data (including alerts, saved objects, etc), or something in between?

When a rule runs, it needs to query source events and write detection alerts. Sometimes, if we talk about "rules on alerts" feature, some rules called "building block rules" would query source events and write alerts which would be used by other rules to generate other alerts. So where would we store all of that in the CCS mode:

- initial source events
- "building block" alerts
- final alerts
- "building block" rules
- final rules

Is this flexible so we could decide where to keep which data per each test, or is it kinda fixed?

MadameSheema replied:
@banderror when we talk about data, we are talking about all the initial setup you need to execute the tests, that could be of any kind, alerts, rules.... and so on.

In the automation we are working on, we are ingesting the data on the remote and querying from the source, since ingesting and querying the data from the same cluster would equal to a regular instance testing. I'm not sure if I'm following you about being flexible.

banderror replied:
@MadameSheema 

> In the automation we are working on, we are ingesting the data on the remote

So here by "data" do you only mean source events in `logs-*`, `metrics-*` etc indices? Or something else too?

> and querying from the source

Does that mean you create detection rules in the "local" cluster, and then these rules query source events in the "remote" cluster? In this case, rules would be Kibana Saved Objects stored in the "local" cluster.

What I was trying to say, in general users can have complex deployments of Elastic stack. E.g. a user could have "building block" rules and alerts stored in the "remote" cluster, and then they could have normal, final rules in the "local cluster". I don't know if there are plans to support these kind of complex setups from the product perspective, but I was thinking it's worth asking if the testing implementation for CCS allows to simulate and test such complex setups (this is what I meant by "flexible") where some data and saved objects could be in one cluster, and some - in another cluster.

banderror commented on x-pack/plugins/security_solution/cypress/README.md:
````diff
@@ -115,8 +115,42 @@ cd x-pack/plugins/security_solution
+yarn cypress:run:ccs
+```
+
+Similar sequence, just ending with `yarn cypress:open:ccs`, can be used for interactive test running via Cypress UI.
+
+Appending `--browser firefox` to the `yarn cypress:run:ccs` command above will run the tests on Firefox instead of Chrome.
````
Just to double-check, is `ccs` a test run mode for running _all_ and _any_ Cypress tests, or only a subset of tests? I guess it only runs tests in the `ccs_integration` folder, but want to make sure this is a correct understanding.

Also, does the normal test run exclude `ccs_integration` folder?

In other words, are "normal" and "ccs" test runs _completely_ separate in terms of tests and configs?

MadameSheema replied:
@banderror the CCS mode executes all the tested inside the `ccs_integration` folder and the normal execution would execute that the tests inside the `integration` folder. So yes, are completely separate runs in terms test, configs and also when those tests are going to be executed.

We have separated the tests  because what are we going to check on CCS is not exactly the same as what we check on a regular scenario as well as the data, the data is going to differ.

banderror commented on x-pack/plugins/security_solution/cypress/README.md:
````diff
@@ -208,6 +242,44 @@ Because of `cy.exec`, used to invoke `es_archiver`, it's necessary to override i
+
+Incorrect handling of the above points might result in false positives, in that the remote cluster is not involved but the test passes anyway.
+
+#### Remote data loading
+
+Helpers `esArchiverCCSLoad` and `esArchiverCCSUnload` are provided by [cypress/tasks/es_archiver.ts](https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/cypress/tasks/es_archiver.ts):
+
+```javascript
+import { esArchiverCCSLoad, esArchiverCCSUnload } from '../../tasks/es_archiver';
+```
+
+They will use the `CYPRESS_CCS_*_URL` environment variables for accessing the remote cluster. Complex tests involving local and remote data can interleave them with `esArchiverLoad` and `esArchiverUnload` as needed.
````
Is my understanding correct that `esArchiverCCSLoad, esArchiverCCSUnload` can be used to work with any es archive (including already existing archives)? The only difference that they work with the remote cluster instead of the local one.

MadameSheema replied:
@banderror that is correct, you can use any archive and the data is going to be loaded on the remote.

banderror reviewed (commented):
Thank you for working on that @cavokz! I left some comments that are mostly questions.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136499/)
* Commit: 32cda59cc0c49bb9d17d46b06253e4b50f08373d
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103941/32cda59cc0c49bb9d17d46b06253e4b50f08373d)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #136490](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136490/) succeeded 985083c14302683b1571565558719e8b84c64e01
* :green_heart: [Build #136441](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136441/) succeeded 62bab351f9a92a970d177f9d181a6423d47acb87
* :broken_heart: [Build #136438](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136438/) failed 10d4bdc71a93d03a2cb445afd6dc3407a022f453
* :green_heart: [Build #136391](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136391/) succeeded 8b021641bc1c35446181c1d9ebaf57e1160f407d
* :green_heart: [Build #136382](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/) succeeded 627ea9b77540839edcc77159a160fa14d6526f25

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136382/","number":"136382","commit":"627ea9b77540839edcc77159a160fa14d6526f25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136391/","number":"136391","commit":"8b021641bc1c35446181c1d9ebaf57e1160f407d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136438/","number":"136438","commit":"10d4bdc71a93d03a2cb445afd6dc3407a022f453"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136441/","number":"136441","commit":"62bab351f9a92a970d177f9d181a6423d47acb87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136490/","number":"136490","commit":"985083c14302683b1571565558719e8b84c64e01"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136499/","number":"136499","commit":"32cda59cc0c49bb9d17d46b06253e4b50f08373d"}]}
PIPELINE-->

MadameSheema commented on x-pack/plugins/security_solution/cypress/README.md:
```diff
@@ -115,8 +115,42 @@ cd x-pack/plugins/security_solution
+# bootstrap Kibana from the project root
+yarn kbn bootstrap
+
+# launch the Cypress test runner with overridden environment variables
+cd x-pack/plugins/security_solution
+CYPRESS_ELASTICSEARCH_USERNAME="user" \
+CYPRESS_ELASTICSEARCH_PASSWORD="pass" \
+CYPRESS_BASE_URL="https://user:pass@server1:5601" \
+CYPRESS_ELASTICSEARCH_URL="https://user:pass@server1:9200" \
+CYPRESS_CCS_KIBANA_URL="https://user:pass@server2:5601" \
+CYPRESS_CCS_ELASTICSEARCH_URL="https://user:pass@server2:9200" \
+CYPRESS_CCS_REMOTE_NAME="remote" \
+yarn cypress:run:ccs
```
@banderror the configuration mentioned in that piece of the README is in case you want to use your own CCS setup, right now if you want to execute the automation from your local, you need to setup the CCS configuration and pass the variables. Doing something like `yarn cypress:run-as-ci` command that starts an setups an instance for you, it is out of scope for now for this PR.  But of course, these are just the first steps and we will continue working to make everything easier to setup :)

cavokz replied:
@banderror once https://github.com/elastic/integration-test/pull/489 is merged you can use [integration-test/jenkins_test.sh](https://github.com/elastic/integration-test/blob/master/jenkins_test.sh) in this way:

```shell
VMS=ubuntu16_tar_ccs_cypress ./jenkins-test.sh
```

It sets up the two clusters with CCS and runs the Cypress CCS tests. The two clusters are in the same VM, managed with vagrant, and a snapshot is taken before the tests are executed. You can always restore the snapshot and run your tests again.

If your tests clean up after themselves, you don't need to restore the snapshot; just re-run `yarn cypress:run:ccs`.

banderror replied:
@MadameSheema gotcha, sounds good. Definitely would be nice to have something like `yarn cypress:run-as-ci`, but for CCS, for local development.

spalger reviewed (approved):
Operations related changes LGTM

kibanamachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.x](https://github.com/elastic/kibana/pull/104496)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/104496">](https://github.com/elastic/kibana/pull/104496) |

This backport PR will be merged automatically after passing CI.

kibanamachine commented:
## 💔 Backport failed

| Status | Branch | Result |
|:------:|:------:|:------:|
| ❌ |  7.14  | Commit could not be cherrypicked due to conflicts |

To backport manually run:
`node scripts/backport --pr 103941`

kibanamachine commented:
## 💔 Backport failed

| Status | Branch | Result |
|:------:|:------:|:------:|
| ❌ |  7.14  | Commit could not be cherrypicked due to conflicts |

To backport manually run:
`node scripts/backport --pr 103941`
